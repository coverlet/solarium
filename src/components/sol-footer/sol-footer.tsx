import React, { ReactElement } from 'react';
import styles from './sol-footer.module.scss';

export const SolFooter = (): ReactElement => {
  return <div className={styles['footer']}>&copy; 2021</div>;
};
