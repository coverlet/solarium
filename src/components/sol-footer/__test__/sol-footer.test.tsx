import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';
import { SolFooter } from '../sol-footer';

// more info
// https://testing-library.com/docs/react-testing-library/example-intro
// https://www.leighhalliday.com/introduction-react-testing-library

afterEach(cleanup);

describe('footer components', () => {
  it('renders', () => {
    const { asFragment } = render(<SolFooter />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('contains text', () => {
    render(<SolFooter />);
    // screen.debug();
    expect(screen.getByText(/Â© 2021/i)).toBeInTheDocument();
  });
});
