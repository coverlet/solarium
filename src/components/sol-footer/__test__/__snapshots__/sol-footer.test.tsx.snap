// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer components renders 1`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    Â© 2021
  </div>
</DocumentFragment>
`;
